File: src/main/java/io/saso/dash/services/ServiceFactory.java
package io.saso.dash.services;

import com.google.inject.name.Named;
import io.saso.dash.database.entities.LiveToken;

public interface ServiceFactory
{
    DBEntityProvider createDBEntityProvider(LiveToken liveToken);

    @Named("github")
    Service createGitHubService();

    @Named("google")
    Service createGoogleService();
}

File: src/main/kotlin/io/saso/dash/services/ServiceFactory.kt
package io.saso.dash.services

import com.google.inject.name.Named
import io.saso.dash.database.DBEntityProvider
import io.saso.dash.database.entities.DBLiveToken

public interface ServiceFactoryOLD
{
    fun createDBEntityProvider(liveToken: DBLiveToken): DBEntityProvider

    @Named("github")
    fun createGitHubService(): Service

    @Named("google")
    fun createGoogleService(): Service
}
